clients/spark/core/src/main/scala/io/treeverse/clients/GCBackupAndRestore.scala

```diff
@@ -0,0 +1,165 @@
+package io.treeverse.clients
+
+import org.apache.hadoop.conf.Configuration
+
+import scala.util.control.Breaks._
+import org.apache.hadoop.tools.DistCp
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.apache.spark.sql.{DataFrame, Dataset, SparkSession}
+
+import java.net.URI
+
+class GCBackupAndRestore {}
+
+object GCBackupAndRestore {
+  val S3AccessKeyName = "fs.s3a.access.key"
+  val S3SecretKeyName = "fs.s3a.secret.key"
+  val AzureStorageAccountKeyName = "fs.azure.account.key"
+  lazy val spark = SparkSession.builder().appName("GCBackupAndRestore").getOrCreate()
+  import spark.implicits._
+
+  def constructAbsoluteObjectPaths(
+      objectsRelativePathsDF: DataFrame,
```

Comment:
These are physical addresses, right?
So where do I get this list in the first place? How do I know which files I want to back up?

Reply:
Right, our docs (that I'll add separately) will guide you on what to look for. 
To backup: you will need to find the addresses created by a mark_only gc run you've conducted before the backup. 
To restore: you will need to find the addresses created by a mark_only gc run you've swept the objects it pointed to, this is what you would want to restore.